core/src/test/scala/unit/kafka/raft/RaftManagerTest.scala

```diff
@@ -34,21 +34,30 @@ import org.mockito.Mockito._
 
 class RaftManagerTest {
 
-  private def instantiateRaftManagerWithConfigs(processRoles: String, nodeId:String) = {
+  private def instantiateRaftManagerWithConfigs(topicPartition: TopicPartition, processRoles: String, nodeId: String) = {
```

Comment:
The issue is that after every test you need to delete the directory created. This is where raft creates the directory
https://github.com/apache/kafka/blob/9bc45d4e031bef63772b3d8d7c5828c0733a27b0/core/src/main/scala/kafka/raft/RaftManager.scala#L212-L215

So you need to override that with a temp dir from https://github.com/apache/kafka/blob/9bc45d4e031bef63772b3d8d7c5828c0733a27b0/core/src/test/scala/unit/kafka/utils/TestUtils.scala#L117
and delete it after every test.

Reply:
I did this using 

`scala.reflect.io.Directory.deleteRecursively`

as you were reviewing.

Reply:
I'll switch it to use `tempDir` shortly.

Reply:
Would this class need to extend KafkaRaftManager to override the directory field in that class with tempDir?